### PR TITLE
workflow barrel track fixes for o2-eve

### PR DESCRIPTION
**Pull request by jmyrcha**

### CODE REVIEW

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h [hunk 1]
```diff
@@ -74,15 +74,20 @@ class VisualisationEvent
   }
   void remove_last_track() { mTracks.pop_back(); } // used to remove track assigned optimistically
 
-  // Adds visualisation cluser inside visualisation event
-  //void addCluster(const VisualisationCluster& cluster)
-  //{ mClusters.push_back(cluster); }
+  // Adds visualisation cluster inside visualisation event
   VisualisationCluster& addCluster(double XYZ[])
   {
     mClusters.emplace_back(XYZ);
```
Comment (shahor02) [inline EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h:81]: what is the reason to use doubles for VisualisationCluster? You just inflate the memory and jsons files (unless limiting precision of the output).

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h [hunk 2]
```diff
@@ -53,21 +55,22 @@ class VisualisationTrack
...
-    double energy;
-    int ID;
+    //  double energy;
+    //  int ID;
     int PID;
-    double mass;
-    double signedPT;
+    //  double mass;
+    //  double signedPT;
     double startXYZ[3];
```
Comment (shahor02) [inline EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h]: same here, float would be fully enough for all these data members.

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h [hunk 3]
```diff
@@ -53,21 +55,22 @@ class VisualisationTrack
...
     double theta;
-    double helixCurvature;
-    int type;
-    ETrackSource source;
+    //  double helixCurvature;
+    //  int type;
+    o2::dataformats::GlobalTrackID::Source source;
   };
   // Constructor with properties initialisation
   VisualisationTrack(const VisualisationTrackVO vo);
```
Comment (shahor02) [inline EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h]: pass by reference

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h [hunk 4]
```diff
@@ -105,22 +110,23 @@ class VisualisationTrack
   void addMomentum(const double pxpypz[3]);
 
   int mID;                     /// Unique identifier of the track
-  std::string mType;           /// Type (standard, V0 mother, daughter etc.)
+                               //  std::string mType;           /// Type (standard, V0 mother, daughter etc.)
+  float mTime;                 /// track time
   int mCharge;                 /// Charge of the particle
-  double mEnergy;              /// Energy of the particle
-  int mParentID;               /// ID of the parent-track (-1 means no parent)
+                               //  double mEnergy;              /// Energy of the particle
```
Comment (shahor02) [inline EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h]: if the commented out code is obsolete, better to remove it.

File: EventVisualisation/Workflow/src/EveWorkflowHelper.cxx [hunk 5]
```diff
@@ -13,167 +13,205 @@
...
-        auto pattIt = patterns.begin();
-        retVal->ITSClustersArray.reserve(clusITS.size());
-        o2::its::ioutils::convertCompactClusters(clusITS, pattIt, retVal->ITSClustersArray, *calib->itsPatternDict);
-      }
+  auto creator = [maskTrk, this](auto& trk, GID gid, float time, float) {
+    if (!maskTrk[gid.getSource()]) {
+      return true;
     }
+    mTrackSet.trackGID.push_back(gid);
+    mTrackSet.trackTime.push_back(time);
```
Comment (shahor02) [inline EventVisualisation/Workflow/src/EveWorkflowHelper.cxx:36]: note that for TPC tracks the time is in timeBins = 8 bunch-crossings = `8*o2::constants::lhc::LHCBunchSpacingMUS` in \mus

File: EventVisualisation/Workflow/src/EveWorkflowHelper.cxx [hunk 6]
```diff
@@ -13,167 +13,205 @@
...
+  drawITSClusters(gidITS, trackTime);
+  drawTPCClusters(gidTPC, trackTime);
+}
+
+std::vector<PNT> EveWorkflowHelper::getTrackPoints(const o2::track::TrackPar& trc, float minR, float maxR, float maxStep)
+{
+  // adjust minR according to real track start fro track starting point
+  float rMin = std::sqrt(trc.getX() * trc.getX() + trc.getY() * trc.getY());
+  if (rMin > minR)
+    minR = rMin;
```
Comment (shahor02) [inline EventVisualisation/Workflow/src/EveWorkflowHelper.cxx]: out codechecker requires  `{}` on different lines even for potential oneliners

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h (hunk 1 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h:81]: json files store float. But you are right - I will change to float.

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h (hunk 2 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h]: done

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h (hunk 3 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h]: it is uint8 - why reference? It is value object used to construct VisualisationTrack - so I guess should contains values

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h (hunk 4 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationTrack.h]: I remove it (but it return in a future)

File: EventVisualisation/Workflow/src/EveWorkflowHelper.cxx (hunk 6 above)
Comment (jmyrcha) [reply EventVisualisation/Workflow/src/EveWorkflowHelper.cxx]: done

File: EventVisualisation/Workflow/src/EveWorkflowHelper.cxx (hunk 5 above)
Comment (jmyrcha) [reply EventVisualisation/Workflow/src/EveWorkflowHelper.cxx:36]: ok - fixed for TPC (during drawing) - but for joined ITSTPC ... I assume time is in ms

**Comment by davidrohr** [comment]
@jmyrcha : Could you please run clang-format?

**Pull request merged**